File: server/src/models/Student.js
const mongoose = require('mongoose');

const studentSchema = new mongoose.Schema({
  user: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  studentId: {
    type: String,
    required: true,
    unique: true,
    trim: true
  },
  class: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Class',
    required: [true, 'Class assignment is required']
  },
  dob: {
    type: Date,
    required: [true, 'Date of birth is required']
  },
  gender: {
    type: String,
    enum: ['Male', 'Female'],
    required: [true, 'Gender is required']
  },
  enrollmentDate: {
    type: Date,
    default: Date.now
  },
  parentName: {
    type: String,
    required: [true, 'Parent/Guardian name is required']
  },
  parentPhone: {
    type: String,
    required: [true, 'Parent/Guardian phone number is required (e.g. +2567...)']
  },
  parentEmail: {
    type: String,
    trim: true
  },
  parentUser: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    default: null
  },
  address: {
    type: String,
    trim: true
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Student', studentSchema);

File: server-springboot/src/main/java/com/ndugu/schoolsystem/model/Student.java
package com.ndugu.schoolsystem.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "students")
public class Student {

    @Id
    private String id;

    private String user; // Reference to User.id (ObjectId represented as String)

    @Indexed(unique = true)
    private String studentId;

    private String clazz; // 'class' is a reserved keyword in Java; reference to Class.id (represented as String)

    private LocalDate dob;

    private String gender; // 'Male', 'Female'

    @Builder.Default
    private LocalDate enrollmentDate = LocalDate.now();

    private String parentName;

    private String parentPhone;

    private String parentEmail;

    private String parentUser; // Reference to parent User.id (represented as String)

    private String address;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
